testutil/parallel.go

```diff
@@ -0,0 +1,17 @@
+package testutil
+
+import "testing"
+
+// IsParallel determines whether the current test is running with parallel set.
+// The Go standard library does not currently expose this. However, we can easily
+// determine this using the fact that a call to t.Setenv() after t.Parallel() will panic.
+func IsParallel(t testing.TB) (parallel bool) {
+	t.Helper()
+	defer func() {
+		if r := recover(); r != nil {
+			parallel = true
+		}
+	}()
+	t.Setenv(t.Name()+"_PARALLEL", "")
```

Comment:
Maybe called `t.Parallel()` again would be more idiomatic?

```go
// Parallel signals that this test is to be run in parallel with (and only with)
// other parallel tests. When a test is run multiple times due to use of
// -test.count or -test.cpu, multiple instances of a single test never run in
// parallel with each other.
func (t *T) Parallel() {
	if t.isParallel {
		panic("testing: t.Parallel called multiple times")
	}
```

Reply:
Wouldn't that cause the test to become parallel? You don't necessarily want that.

Reply:
Right, whoops.